moveit_common/cmake/moveit_package.cmake

```diff
@@ -41,6 +41,20 @@ macro(moveit_package)
   set(CMAKE_CXX_STANDARD_REQUIRED ON)
   set(CMAKE_CXX_EXTENSIONS OFF)
 
+  if(NOT DEFINED ENV{ROS_DISTRO})
```

Comment:
FYI: this breaks the moveit2_packages: https://github.com/moveit/moveit2_packages/actions/runs/4749763095/jobs/8437320692

The better way would be to check the rclcpp version where this change was introduced.